js/data/symbol_bucket.js

```diff
@@ -36,10 +36,16 @@ function SymbolBucket(options) {
     var zoomHistory = { lastIntegerZoom: Infinity, lastIntegerZoomTime: 0, lastZoom: 0 };
 
     this.adjustedTextMaxSize = this.layer.getLayoutValue('text-size', 18, zoomHistory);
-    this.adjustedTextSize = this.layer.getLayoutValue('text-size', this.zoom + 1, zoomHistory);
+    this.adjustedTextSizes = [
+        this.layer.getLayoutValue('text-size', this.zoom, zoomHistory),
+        this.layer.getLayoutValue('text-size', this.zoom + 1, zoomHistory)
+    ];
 
     this.adjustedIconMaxSize = this.layer.getLayoutValue('icon-size', 18, zoomHistory);
-    this.adjustedIconSize = this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory);
+    this.adjustedIconSizes = [
+        this.layer.getLayoutValue('icon-size', this.zoom, zoomHistory),
+        this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory)
+    ];
```

Comment:
This isn't an ideal general-case solution because zoom stops can be at fractional zooms. How would you feel about adding a `getMaxLayoutValue(name, minZoom, maxZoom, zoomHistory)` that does a little more function introspection?